tests/rados/test_pool_osd_recovery.py

```diff
@@ -40,312 +41,409 @@ def run(ceph_cluster, **kw) -> int:
+                "New hosts added to the cluster successfully, Proceeding to deploy OSDs on the same."
+            )
+            # Deploying OSDs on the new nodes.
+            osd_args = {
+                "steps": [
+                    {
+                        "config": {
+                            "command": "apply_spec",
+                            "service": "orch",
+                            "validate-spec-services": True,
+                            "specs": [
+                                {
+                                    "service_type": "osd",
+                                    "service_id": "new_osds",
+                                    "encrypted": "true",
+                                    "placement": {"label": "osd-bak"},
+                                    "spec": {"data_devices": {"all": "true"}},
+                                }
+                            ],
+                        }
+                    }
+                ]
+            }
+            osd_args.update(config)
+            ocount_pre = len(get_osd_details(node=cephadm))
+            add_osd(ceph_cluster=ceph_cluster, config=osd_args)
+            if not ocount_pre <= len(get_osd_details(node=cephadm)):
+                log.error("New OSDs were not added into the cluster")
+                raise Exception("Execution error")
+
+            log.info("Deployed new hosts and deployed OSDs on them")
+        except Exception as e:
+            log.error(f"Failed with exception: {e.__doc__}")
+            log.exception(e)
+            return 1
+
+    def remove_custom_host(host_node_name: str):
+        try:
+            # Removing newly added OSD host and checking status
+            rm_host = get_node_by_id(ceph_cluster, host_node_name)
+            log.info(
+                f"Identified host : {rm_host.hostname} to be removed from the cluster"
+            )
 
-    # Checking cluster health before starting the tests
-    method_should_succeed(rados_obj.run_pool_sanity_check)
-    log.info(
-        "Completed health check of the cluster after test pool creation. Cluster health good!!"
-    )
+            # get list of osd_id on the host to be removed
+            rm_osd_list = rados_obj.get_osd_df_stats(
+                tree=True, filter_by="name", filter=rm_host.hostname
+            )["nodes"][0]["children"]
+            dev_path_list = []
+            for osd_id in rm_osd_list:
+                dev_path_list.append(get_device_path(host=rm_host, osd_id=osd_id))
+
+            # Starting to drain the host
+            drain_cmd = f"ceph orch host drain {rm_host.hostname}"
+            cephadm.shell([drain_cmd])
+            # Sleeping for 2 seconds for removal to have started
+            time.sleep(2)
+            log.debug(f"Started drain operation on node : {rm_host.hostname}")
+
+            status_cmd = "ceph orch osd rm status -f json"
+            end_time = datetime.datetime.now() + datetime.timedelta(seconds=3000)
```

Comment:
Since we have smart wait, can we increase the timeout to 600?

Reply:
There is never any guarantee for the drain time taken by host, esp when we run it on BM clusters.